docs/week-3.md

```diff
@@ -0,0 +1,523 @@
+
+### useState Hook
+`useState` allows us to add and keep track of data in the state. In order to use this hook we must import 
+it from react. We use {} when importing useState because it is not a default export but a named one. We will
+discuss default and named exports in Week 6.
+
+```import { useState } from 'react';```
+
+Let us add that import into John's profile component.
+
+```
+import React, { useState} from 'react';
+import LuckyNum from './LuckyNum';
+
+const Profile = () => (
+
+    <div>
+        <img src="profileImage.jpg"/>
+        <h1>Hi, my name is John! My favorite color is blue.</h1>
+        <LuckyNum />
+        <button>Likes: </button>
+    </div>
+);
+
+export default Profile;
+```
+
+Now in order to use `useState` we make a `const` array with the a variable and a setter. We then set this equal
+to a call to useState and pass in the type of data we want to keep track of. For example if we want to keep 
+track of someones message we would write this:
+
+```const [message, setMessage] = useState("Hello World")```
+
+We have initialized `message` to be "Hello World". If we want to change our message we would simply use 
+`setMessage`.
+
+```setMessage("This is our new message")```
+
+Now John can simply implement `useState` into his profile.
+
+```
+import React, { useState } from 'react';
+import LuckyNum from './LuckyNum';
+
+const Profile = () => (
+
+    //initialize the state to 0 because we start with 0 likes
+    const [profileLikes, setProfileLikes] = useState(0);
+
+    <div>
+        <img src="profileImage.jpg"/>
+        <h1>Hi, my name is John! My favorite color is blue.</h1>
+        <LuckyNum />
+        <button>Likes: {profileLikes}</button>
+    </div>
+);
+
+export default Profile;
+```
+
+Now all we have to do is create an `onClick` event to know when to increment the likes.
+
+```
+import React, { useState } from 'react';
+import LuckyNum from './LuckyNum';
+
+const Profile = () => (
+
+    //initialize the state to 0 because we start with 0 likes
+    const [profileLikes, setProfileLikes] = useState(0);
+
+    //handles incrementing the likes of John's profile
+    function handleLikes {
+        setProfileLikes(profileLikes + 1);
+    }    
+
+    <div>
+        <img src="profileImage.jpg"/>
+        <h1>Hi, my name is John! My favorite color is blue.</h1>
+        <LuckyNum />
+        <button onClick="handleLikes()">Likes: {profileLikes}</button>
+    </div>
+);
+
+export default Profile;
+```
+
+## Typing with Typescript
+As you may have noticed, react is normally implement with Javascript. The only downside to using Javascript is 
+that we do not get type enforcement.
+
+### Why use Typescript in the first place?
+When developing an application with a lot of contributors it is important to have some sort of structure to 
+your code. Typing allows us to not create errors and mess up a project with other contributors on it. Typescript
```

Comment:
Typing supposed to be TypeScript?

Reply:
Typing is the act of giving our code types. I will include that.